src/trans.cc

```diff
@@ -1486,38 +1382,32 @@ static Obj FuncSMALLEST_IDEM_POW_TRANS(Obj self, Obj f)
 ** GAP level functions for regularity of transformations
 *******************************************************************************/
 
-// Returns True if the transformation or list <t> is injective on the list
-// <l>.
+// Returns True if the transformation or list <obj> is injective on the list
+// <list>.
 
-static Obj FuncIsInjectiveListTrans(Obj self, Obj l, Obj t)
+static Obj FuncIsInjectiveListTrans(Obj self, Obj list, Obj obj)
 {
     UInt    n, i, j;
     const UInt2 * ptt2;
     const UInt4 * ptt4;
     UInt4 * pttmp = 0L;
     Obj     val;
 
-    if (!IS_LIST(l)) {
-        ErrorQuit("the first argument must be a list (not a %s)",
-                  (Int)TNAM_OBJ(l), 0L);
-    }
-    else if (!IS_TRANS(t) && !IS_LIST(t)) {
-        ErrorQuit("the second argument must be a transformation or a list "
-                  "(not a %s)",
-                  (Int)TNAM_OBJ(t), 0L);
+    RequireSmallList("IsInjectiveListTrans", list);
+    if (!IS_TRANS(obj) && !IS_LIST(obj)) {
+        RequireArgument("IsInjectiveListTrans", obj, "must be a transformation or a list");
     }
     // init buffer
-    n = (IS_TRANS(t) ? DEG_TRANS(t) : LEN_LIST(t));
+    n = (IS_TRANS(obj) ? DEG_TRANS(obj) : LEN_LIST(obj));
     pttmp = ResizeInitTmpTrans(n);
 
-    if (TNUM_OBJ(t) == T_TRANS2) {
-        ptt2 = CONST_ADDR_TRANS2(t);
-        for (i = LEN_LIST(l); i >= 1; i--) {
-            val = ELM_LIST(l, i);
+    if (TNUM_OBJ(obj) == T_TRANS2) {
+        ptt2 = CONST_ADDR_TRANS2(obj);
+        for (i = LEN_LIST(list); i >= 1; i--) {
+            val = ELM_LIST(list, i);
             if (!IS_POS_INTOBJ(val)) {
                 ErrorQuit(
-                    "the entries of the first argument must be positive "
-                    "integers (not a %s)",
+                    "<list> must be a list of positive small integers (not a %s)",
                     (Int)TNAM_OBJ(val), 0L);
```

Comment:
Hmm, I think I may have accidentally made the error message worse. To fix this, I should either drop the `(not a %s)`, as the messages makes it sound as if this refers to the overall list; or perhaps it could be something like this:
```C
  ErrorQuit("IsInjectiveListTrans: <list>[%d] must be a positive small integer (not a %s)", i, (Int)TNAM_OBJ(val));
```

Reply:
I have a new PR ready which fixes this. I think it would be easiest to just merge this PR here as it is, even though that error above is flawed; and then I'd immediately submit a new PR which replaces this bit code with an IMHO much better solution.

Reply:
Fine by me.